src/main/resources/templates/fragments/common.html

```diff
@@ -5,29 +5,40 @@
     <title>Title</title>
 </head>
 <body>
-<div th:fragment="nav-loggedout">
-    <nav>
-        <a href="/">Home</a>
-        <a href="/login">Login</a>
-        <a href="/signup">Join</a>
-    </nav>
-</div>
-
-<div th:fragment="nav-loggedin-user">
-    <nav>
+<div th:fragment="navbar">
+    <nav th:if="${username}">
         <a href="/">Home</a>
+        <a th:href="'/myprofile/'+${userID}">My Profile</a>
         <a href="/logout">Logout</a>
     </nav>
-</div>
 
-<div th:fragment="nav-loggedin-admin">
-    <nav>
+    <nav th:unless="${username}">

```

Comment:
Excellent use of fragments, and Thymeleaf conditionals... One nav for logged in users, and one for non-logged in 👍👍